src/internal.coffee

```diff
@@ -14,6 +20,85 @@ module.exports =
       throw new Error "Neither Blob nor Buffer are accessible in this environment. " +
         "Consider adding Blob.js shim"
 
+  generateMHTDocument: (htmlSource) ->
+    # take care of images
+    @prepareImageParts(htmlSource).then (result) ->
+      if result.htmlSource
+        {htmlSource, imageContentParts} = result
+      else
+        htmlSource = htmlSource.outerHTML
+        imageContentParts = result.imageContentParts
+      # for proper MHT parsing all '=' signs in html tags need to be replaced with '=3D'
+      htmlSource = htmlSource.replace /\=/g, '=3D'
+      W.resolve mhtDocumentTemplate {htmlSource, contentParts: imageContentParts.join '\n'}
+    , (err) -> throw err
+
+  prepareImageParts: (htmlSource) ->
```

Comment:
I would prefer to have all this image manipulation code in a different file (if we keep it), it is somewhat auxiliary to the main library objective.